cvat/apps/dataset_manager/bindings.py

```diff
@@ -2023,25 +2023,32 @@ def _clean_display_message(self) -> str:
             message = "Dataset must contain a file:" + message
         return re.sub(r' +', " ", message)
 
-def mangle_image_name(name: str, subset: str, names: defaultdict[tuple[str, str], int]) -> str:
-    name, ext = name.rsplit(osp.extsep, maxsplit=1)
+def mangle_image_name(name: str, subset: str, names: defaultdict[tuple[str, str], int], task_id: Optional[int] = None) -> str:
+    # Track the base name and extension of the image file
+    name_base, ext = name.rsplit(osp.extsep, maxsplit=1)
+
+    if not names[(subset, name_base)]:
+        names[(subset, name_base)] += 1
+        return osp.extsep.join([name_base, ext])
+
+    if task_id is not None:
```

Comment:
Seems unnecessary complex. Why not just append the task id if it exists and keep the original logic?